src/fsharp/ast.fs

```diff
@@ -804,6 +804,9 @@ and
     /// 'use x = fixed expr'
     | Fixed of expr: SynExpr * range: range
 
+    /// Reverse index in slices
+    | ReverseIndex of expr: SynExpr * opRange: range * range: range
```

Comment:
I think we can avoid having to add a case to `SynExpr`.
The operator `^` to denote `ReverseIndex` can only happen in the context of the indexer. It's not a general concept that any expression can have thus it doesn't belong as a new case on `SynExpr`.

Instead, I recommend tweaking `SynIndexerArg`:
```fsharp
    SynIndexerArg =

    | Two of SynExpr * fromEnd1: bool * SynExpr * fromEnd2: bool * range

    | One of SynExpr * fromEnd: bool * range

    member x.Range = match x with Two (_, _, _, _, m) | One (_, m) -> m

    member x.Exprs = match x with Two (e1, _, e2, _, _) -> [e1;e2] | One (e, _) -> [e]
```
The addition of `range` is to say we captured the full range of the indexer arg at parse time, which will include any leading `^` token's range. 

What are your thoughts?

Reply:
This is a good idea and should be possible. We'd just have to remove the logic inside the parser to generate `SynIndexerArg.One(Tuple(...))` for `.[1,1]`.